apis/clusters/v1beta1/cassandra_types.go

```diff
@@ -52,38 +51,142 @@ type CassandraRestoreFrom struct {
 
 // CassandraSpec defines the desired state of Cassandra
 type CassandraSpec struct {
-	RestoreFrom         *CassandraRestoreFrom `json:"restoreFrom,omitempty"`
-	OnPremisesSpec      *OnPremisesSpec       `json:"onPremisesSpec,omitempty"`
-	Cluster             `json:",inline"`
+	GenericClusterSpec `json:",inline"`
```

Comment:
It is not available to do this thing with `ClusterStatus` because it still exists in the code. So for now I want to keep it as it is. When we refactor all the cluster resources to this pattern it will be changed too

Reply:
ok